lib/modules/manager/cargo/artifacts.ts

```diff
@@ -120,14 +121,23 @@ async function updateArtifactsImpl(
     if (isLockFileMaintenance) {
       await cargoUpdate(packageFileName, true, config.constraints?.rust);
     } else {
-      const missingDep = updatedDeps.find((dep) => !dep.lockedVersion);
-      if (missingDep) {
-        // If there is a dependency without a locked version then log a warning
-        // and perform a regular workspace lockfile update.
-        logger.warn(
-          { dependency: missingDep.depName },
-          'Missing locked version for dependency',
-        );
+      const nonCrateDep = updatedDeps.find(
+        (dep) => dep.datasource !== CrateDatasource.id,
+      );
+      const crateDepWithoutLockedVersion = updatedDeps.find(
+        (dep) => !dep.lockedVersion && dep.datasource === CrateDatasource.id,
+      );
+      // Non-crate dependencies (like git ones) do not have locked versions.
+      // For crate dependencies, a locked version is expected.
+      // In both situations, perform a regular workspace lockfile update.
+      if (nonCrateDep || crateDepWithoutLockedVersion) {
```

Comment:
connected to above suggestion

```suggestion
      if (hasNonCrateDep || crateDepWithoutLockedVersion) {
```